src/main/java/de/tum/in/www1/artemis/service/AuthorizationCheckService.java

```diff
@@ -126,6 +126,11 @@ public boolean isStudentInCourse(Course course, User user) {
      * @return true, if user is student is owner of this participation, otherwise false
      */
     public boolean isOwnerOfParticipation(Participation participation) {
+        // The student is null if we have a template or solution participation -> check for instructor
+        if (participation.getStudent() == null) {
+            return isAtLeastInstructorForExercise(participation.getExercise());
+        }
```

Comment:
this method is used in many other cases, are you sure that there are not side effects with this change?

Reply:
@krusche I've checked the other uses of this methods and participation.getStudent() should not be null on any other exercises except the programming exercises, right? (this means we do not store the solutions etc. of this exercises using a dummy-participation, correct?).

In the cases where this is used in the context of a programming exercise, this should be fine as every participation without student is a template/solution and therefor should only be viewed by instructors or higher.

I can also move the check of `participation.getStudent() == null` into the ResultResource to avoid influencing anything else if this seems safer to you.

Reply:
Yes this would be better. Who knows if someone else uses this method in the future where student can be null as well and then we might have a problem

Reply:
I've moved it.